### DISCUSSION TITLE
Order Value in Euros not included in receipt data - Required for Customs Purposes

### DISCUSSION THREAD

**Discussion by thisisjake** [etsy/open-api#697] [Ideas]
Hello,

When reviewing orders on the website, we can access the following information for EU based customers:

"You must share Etsy's IOSS number, IM000000000, with your courier for this order (use it only for this purpose). This helps ensure that your buyer won't be charged VAT twice. Make sure to write the value of this order in euros, €49.13, on your package, too."

From the above we can see that the order's value in Euros has been recorded as 49.13 however the API does not show this value. Instead we only get the value in our local currency (YEN). We need to list this EXACT value on our customs declaration as it is a legal requirement.

Without this information in the API we are unable to get this information without accessing the order manually. Due to the nature of currency fluctuations, we are obviously unable to just generate this from the value in our local currency. Though even if we do try this, the numbers seem to be a fair bit off.

Can you please advise?

**Comment by etsyachristensen** [comment]
This isn't something that is currently supported. Though we do have this as an active issue here: https://github.com/etsy/open-api/issues/564

We'll close that issue out in favor of tracking this request here. We'll also move this to the Ideas category so it be more easily tracked as a requested feature.

**Comment by scottjlaw** [comment]
I'd like to also add that it'd be very useful if the IOSS number (or other country equivalent) is returned via the API as it can be difficult to keep track of these as new countries move to using similar things.

**Reply by etsyachristensen** [reply]
Unfortunately we are not allowed to add it to the OpenAPI. This has been requested a lot but the security and legal folks say no.

**Reply by scottjlaw** [reply]
Fair enough! I thought I'd mention it as I noticed Amazon's new API does, and Etsy email notifications do. Perhaps a field indicating that a tax ID is required but not mentioning the number? That'd be particularly useful in situations where it specifically shouldn't be included (i.e I believe UK orders over a certain amount). There can be confusion as an EU order for instance returns a value in the total_tax_cost field, but not the total_vat_cost field. 

Can be difficult for people to keep up with the changes in various countries as it's all got quite complicated!

**Comment by sowsinsk** [comment]
While it may not be possible to add the IOSS number itself to the API results, it would be extremely useful to know that VAT was collected, and the amount in Euros (or other country equivalent). Right now, there is no way for us to tell via the API that special handling is needed on the order (passing the IOSS number along to the shipping carrier, etc).

**Reply by scottjlaw** [reply]
It does include the VAT amount in the total_tax_cost field (but not the total_vat_cost field!). Although that tax field does also include US sales tax where these kinds of rules don't apply, so differentiation would be useful. So you can kind of do that, but you don't know exactly what they were charged in their currency. You could previously work that out via another API call, but that doesn't return it in the correct currency in the new version.

**Comment by thisisjake** [comment]
Completely agree.

**Comment by scottjlaw** [comment]
Is the bug with the V3 API endpoint /v3/application/shops/{shop_id}/receipts/{receipt_id}/payments currencies ever going to be fixed?

i.e I have an order paid in USD with a shop currency of GBP

Order total is £39.60 + Tax of £2.39 = £41.99

On packing slip via the website it is USS$49.81 + Tax US$3.01 = US$52.82

Exchange rate wise it's about right. 

However, the payments endpoint returns (the same for posted as well). It returns the GBP values, but with USD as the currency code. As it stands, this endpoint is pretty useless as these values are incorrect. Currency code either needs to be fixed, or ideally, the values reflect the currency code.

When working correctly, this endpoint can be used to get the correct EUR values for EU imports. It works correctly on V2, but not on V3, and if March 31st is the end date for V2, this needs solving.

```
amount_gross: {
	amount: 4199,
	divisor: 100,
	currency_code: "USD",
},
amount_fees: {
	amount: 188,
	divisor: 100,
	currency_code: "USD",
},
amount_net: {
	amount: 3772,
	divisor: 100,
	currency_code: "USD",
}
```

**Reply by daliuss** [reply]
I have another example of buggy currency date of V3 API endpoint /v3/application/shops/{shop_id}/receipts/{receipt_id}/payments;

The shop currency is in USD and buyer currency is GBP. The order total is $8.00 ($9.60 with tax). 

The message on receipt: 
> Please write Etsy’s UK VAT number, 370 6004 28, on your package (use it only for this purpose). This helps ensure that your buyer won't be charged VAT twice. Make sure to write the value of this order in pounds, **GBP 6.99**, on your package, too.

However Api V3 returns incorrect amount_net of 6.62 instead of 6.99 

```          
           "amount_gross": {
                "amount": 873,
                "divisor": 100,
                "currency_code": "GBP"
            },
            "amount_fees": {
                "amount": 65,
                "divisor": 100,
                "currency_code": "GBP"
            },
            "amount_net": {
                "amount": 662,
                "divisor": 100,
                "currency_code": "GBP"
            },
            "posted_gross": {
                "amount": 873,
                "divisor": 100,
                "currency_code": "GBP"
            },
            "posted_fees": {
                "amount": 65,
                "divisor": 100,
                "currency_code": "GBP"
            },
            "posted_net": {
                "amount": 662,
                "divisor": 100,
                "currency_code": "GBP"
            },
```

**Reply by scottjlaw** [reply]
Weird! So the figures you are getting are neither the USD or GBP amounts??

**Reply by scottjlaw** [reply]
Also it seems that end point, even if they fix the currency values, won't return the correct information for customs forms.

I've just noticed an order that has this "Make sure to write the value of this order in euros, €127.43, on your package, too."

But the customer paid in USD, so their invoice has "US$183.97". The shop currency is in GBP and has £143.99.

The API endpoint is returning amount 14399 and currency USD, so the GBP amount, the buyer currency code, but neither of the two things needed - the customs amount/currency code! 😂

**Comment by possumworx** [comment]
Commenting here to add my support for IOSS info being added to the API. I'm managing to automate so much else of my shop's dispatch process but not this, it's really frustrating!

**Comment by openapi-support** [comment]
We are looking into this issue!
